network/router_test.go

```diff
@@ -451,6 +459,48 @@ func testRouterSendMsgDuplex(t *testing.T, fac routerFactory) {
 	log.Lvl2("Received msg h2 -> h1", msg)
 }
 
+func TestRouterFilterConnectionsIncoming(t *testing.T) {
+	r1, err := NewTestRouterTCP(7878)
+	require.NoError(t, err)
+	r2, err := NewTestRouterTCP(7879)
+	require.NoError(t, err)
+
+	// r1 does not accept any connection
+	go r1.Start(peerIsNeverValid)
+
+	log.OutputToBuf()
+	_, err = r2.Send(r1.ServerIdentity, &SimpleMessage{42})
+	require.Error(t, err)
```

Comment:
If this test fails, then all other tests will have `OutputToBuf`, which is not ideal. Perhaps a `defer log.OutputToOs()`?

Reply:
I have a few more issues with this test:
* The `require` on line #473 fails as is written here when I run it locally (i.e. `err == nil`), but on Travis it usually gets `err != nil` (but not always...).
* Furthermore, having to `Sleep()` and `Stop()` the routers to get the result feels very brittle to me (how long should I `Sleep` for?).

Would you have a suggestion as to how to better write these tests?

Reply:
OK, played around with it a bit, and I think the problem is that the `Router.Send` method does a `r.connect`, which it thinks it succeeds, as the connection is only closed _after_ the handshake.
I tried to close the connection in `Router.receiveServerIdentity`, but that doesn't work, neither. 

So instead of testing the return error of `Router.Send`, which is not well defined, ignore the error from `r2.Send` and check if `r1.connections` has an entry for `r2`.

Reply:
Which means that you have to wait for the error in the `StdErr` of `r1`...

Reply:
Thank you!